src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/TextChangeExtensions.cs

```diff
@@ -1,25 +1,51 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.Text;
+using RoslynTextChange = Microsoft.CodeAnalysis.Text.TextChange;
 
 namespace Microsoft.VisualStudio.LanguageServerClient.Razor
 {
     public static class TextChangeExtensions
     {
-        public static bool IsDelete(this TextChange textChange)
+        public static bool IsDelete(this ITextChange textChange)
         {
-            return textChange.Span.Length > 0 && textChange.NewText.Length == 0;
+            return textChange.OldSpan.Length > 0 && textChange.NewText.Length == 0;
         }
 
-        public static bool IsInsert(this TextChange textChange)
+        public static bool IsInsert(this ITextChange textChange)
         {
-            return textChange.Span.Length == 0 && textChange.NewText.Length > 0;
+            return textChange.OldSpan.Length == 0 && textChange.NewText.Length > 0;
         }
 
-        public static bool IsReplace(this TextChange textChange)
+        public static bool IsReplace(this ITextChange textChange)
         {
-            return textChange.Span.Length > 0 && textChange.NewText.Length > 0;
+            return textChange.OldSpan.Length > 0 && textChange.NewText.Length > 0;
+        }
+
+        public static IReadOnlyList<ITextChange> ToVisualStudioTextChangeList(this IReadOnlyList<RoslynTextChange> roslynTextChanges)
```

Comment:
>this IReadOnlyList<RoslynTextChange> [](start = 78, length = 36)

This seems like an odd thing to add an extension method too. Perhaps just put it on RoslynTextChange?

Reply:
We never need to convert just a single change, always an IReadOnlyList<RoslynTextChange>. If you look at the usage in DefaultRazorLanguageServerCustomMessageTarget, this makes sense I think. Let me know if you disagree and we can chat. 

---
In reply to: [447129760](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447129760) [](ancestors = 447129760)

Reply:
OK, I obviously prefer defining it on the textchange, but I don't care deeply and will let the razor folks chime in if they have a preference.

---
In reply to: [447234965](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447234965) [](ancestors = 447234965,447129760)

Reply:
From a design perspective, it feels weird to extend the functionality of `IReadOnlyList<T> where T : RoslynTextChange` (I'm being verbose here to make a point) rather than extending `RoslynTextChange` itself.  The interesting portion of this method isn't that it operates on the list, but how it converts each individual item (from a more extreme perspective, that ought to be refactored out into its own method anyway), and calling the convertion in a for loop or a `Select()` projection seems trivial.

Just 2¢.  Not feeling strongly here.

Reply:
I've never been a design purist, so extending an IReadOnlyList doesn't bother me since that's 100% of the usage. I like the simplicity of being able to do

request.Changes.ToVisualStudioTextChangeList()

without any extra code (not that extra would be that complicated, just unnecessary). 

---
In reply to: [447239227](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447239227) [](ancestors = 447239227)

Reply:
Ya I think i'd prefer this be on the text change itself in the form of an implicit operator. I forget if the runtime supports implicitly converting arrays... If it did that'd be dope!

Reply:
+1 to not having a list extension method

Reply:
OK, you guys win since you own the code. :)

---
In reply to: [447254298](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447254298) [](ancestors = 447254298)

Reply:
Per offline chat, doing a simple extension method instead of an operator since that would've needed to be defined in VisualStudioTextChange class itself thus making it necessary to either re-add Roslyn reference to that assembly or move that class or do something else. 

---
In reply to: [447244697](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447244697) [](ancestors = 447244697)

Reply:
Still disagree with all four of you :-P 

---
In reply to: [447275834](https://github.com/dotnet/aspnetcore-tooling/pull/2137#discussion_r447275834) [](ancestors = 447275834,447254298)